### PR TITLE
Navigation: Convert `PluginDetails` page to use new `Page` extensions

### LINKED ISSUE #58210: PluginPage: Expose a mechanism for customising the `PageHeader`
- [ ] add documentation for `PluginPage` props

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- converts `PluginDetails` page to use the new `Page` extensions added by https://github.com/grafana/grafana/pull/58465
- removes `headerExtra` - the same can now be achieved by passing a custom `subTitle`

| | before | after |
| --- | --- | --- |
| topnav on | image | image |
| topnav off | image | image |

**Why do we need this feature?**

- consistent page design!

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/58210

**Special notes for your reviewer**:

- page responsiveness is **no worse** than before when topnav is off, should be lots better when topnav is on

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
This is looking great. Just think we need to move some of the messages out of InstallControls

**Comment by ashharrison90** [comment]
@torkelo @grafana/plugins-platform-frontend can i check something - when a plugin has `PluginSignatureType` === `core`, is there any point in showing the downloads number? 🤔 i'm not exactly sure what it corresponds to and it seems way off (243 downloads for prometheus... _ok_)

File: public/app/features/plugins/admin/hooks/usePluginInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  const latestCompatibleVersion = getLatestCompatibleVersion(plugin.details?.versions);
+  const version = plugin.installedVersion || latestCompatibleVersion?.version;
+
+  const actions = (
+    <>
+      <InstallControls plugin={plugin} latestCompatibleVersion={latestCompatibleVersion} />
+      <GetStartedWithPlugin plugin={plugin} />
+    </>
+  );
```
[the remark this answers is not in the document]
Comment (ashharrison90) [reply public/app/features/plugins/admin/hooks/usePluginInfo.tsx:29]: @torkelo where do we think the messages should sit, under the subtitle? 🤔

**Comment by leventebalogh** [comment]
**`headerExtra`**
Although it could be interpreted as a breaking change, I guess it was only [introduced a month ago](https://github.com/grafana/grafana/commit/018733dd24df2c2701ba079f859a29db254ecf1f) it hopefully shouldn't have too many dependant plugins. Do you know about any internal plugin depending on this?

> @torkelo https://github.com/orgs/grafana/teams/plugins-platform-frontend can i check something - when a plugin has anything other than PluginSignatureType === community, is there any point in showing the downloads number? 🤔 i'm not exactly sure what it corresponds to and it seems way off (243 downloads for prometheus... ok)

@ashharrison90 I think it probably makes sense for at least `"community"` and `"commercial"`, but maybe even for `"grafana"` as well (for `"core"` and `"private"` it doesn't makes sense I would think).

File: public/app/features/plugins/admin/hooks/usePluginInfo.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/plugins/admin/hooks/usePluginInfo.tsx:29]: @ashharrison90 yes, that should work better

**Comment by ashharrison90** [comment]
@leventebalogh i don't know of any other internal plugins using `headerExtra`, part of the reason we wanna remove it now before anyone else depends on it 😅

**Review by leventebalogh — APPROVED** [review]
Nice! 🚀 

(I just checked - as much as possible - and couldn't find any plugin depending on `headerExtra` from our catalog, so I guess we should be fine. Brought it up only so we are aware :))

File: public/app/features/plugins/admin/pages/PluginDetails.tsx [hunk 2]
```diff
@@ -27,9 +28,27 @@ export default function PluginDetails({ match, queryParams }: Props): JSX.Elemen
 
   const plugin = useGetSingle(pluginId); // fetches the localplugin settings
   const { navModel, activePageId } = usePluginDetailsTabs(plugin, queryParams.page as PluginTabIds);
+  const { actions, info } = usePluginInfo(plugin);
   const { isLoading: isFetchLoading } = useFetchStatus();
   const { isLoading: isFetchDetailsLoading } = useFetchDetailsStatus();
   const styles = useStyles2(getStyles);
+  const subtitle = (
+    <div className={styles.subtitle}>
+      <div>{plugin?.description}</div>
```
[the remark this answers is not in the document]
Comment (ashharrison90) [reply public/app/features/plugins/admin/pages/PluginDetails.tsx:37]: yep good shout

File: public/app/features/plugins/admin/hooks/usePluginInfo.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashharrison90) [reply public/app/features/plugins/admin/hooks/usePluginInfo.tsx]: @torkelo ok have made that change and refactored a little. it's a bit annoying as the logic for hiding the install buttons/showing the warning message is kind of duplicated between the `InstallControlsWarning` component and the `hasInstallControlWarning` helper, but i couldn't really think of a better way of doing it... open to any ideas 😄

**Review by torkelo — APPROVED** [review]
:tada:

**Comment by teddyba** [comment]
Firstly, thank you so much for working on this @torkelo @ashharrison90 (and anyone else who contributed). I think it is looking really great! 

I have a few questions about the new header: 

1. Is there a component for the tag design that we can reuse for the integration details page? I was hoping we would drive more consistency between this page for data sources (I have been calling it the **data source details page**) and the **integration details page**, like so: 

Screen Shot 2023-01-03 at 8 39 02 PM

Screen Shot 2023-01-03 at 8 39 51 PM


2. Have we identified any rules for how these tags work? These are some of the things I never ironed out with the designs, and was thinking would be a collaborative effort with design/eng/product to determine: 

- How many max tags are shown? 
- If there are more than the max, which tags do we show on the header, and where do the other tags appear? 
- Should we introduce design rules around color / icon usage here?

**Comment by teddyba** [comment]
FYI @mikkancso @jarben

**Comment by ashharrison90** [comment]
hi @teddyba 👋 

1. there's a `PageInfo` component for the actual label/value pairs, but it sounds like you should just be utilising the `Page` component itself. that exposes an `info` prop, which takes an array of label/value pairs where the label is a `string` and the `value` is an arbitrary ReactNode.
2. there's no max right now. realistically, more than 5 or so will start to look a bit weird, but nothing should break. the `Page` component should behave responsibly in all cases. if the number of tags is too big, it'll wrap onto the next line. if it's still overflowing the page bounds, it will scroll horizontally. we could do better at indicating that it's horizontally scrollable (it's not super obvious unless it cuts off half an element or something), but that's quite low on the priority list imo.

https://user-images.githubusercontent.com/20999846/210531302-af534a8b-f959-44cc-945a-acae67a49910.mp4

> Should we introduce design rules around color / icon usage here?

think that's a question for you UXers. 😄 i'm always in favour of design rules as it makes my job as a dev much easier.

**Pull request merged**